internal/iopool/copy.go

```diff
@@ -0,0 +1,26 @@
+package iopool
+
+import (
+	"io"
+	"sync"
+)
+
+type buffer struct {
+	b []byte
+}
+
+var _copyBufSize = 1024 * 32
+var _pool = sync.Pool{
+	New: func() interface{} {
+		return &buffer{make([]byte, _copyBufSize)}
+	},
+}
+
+// Copy wraps the io library's CopyBuffer func with a preallocated buffer from a
+// sync.Pool we maintain.
+func Copy(dst io.Writer, src io.Reader) (int64, error) {
```

Comment:
doc goes too much into internal details; just say,

> Copy copies bytes from the Reader to the Writer until the Reader is exhausted.

And add a non-godoc comment explaining why we have our own.